### PR TITLE
chapitre/tests: add integration tests part

### PR DESCRIPTION
**Pull request by jlevesy**
### What does this PR do?

This PR adds some content for the integration tests part. 

~TODO: find a maven configuration that allows to trigger separately unit tests and integration tests. Or give up on this?~
~TODO: write a part about enabling those tests in the CI.~

### CODE REVIEW

**Comment by jlevesy** [comment]
Okay this is killing me. @dduportal I'm going to need your 🧠 on this. 

Whatever xml crap I throw in my pom.xml. `mvn integration-test`, still. runs. unit-tests before launching actual integration-test.

It  seems to me that `surefire` binds to the `integration-test` phase and I can't find a way to tell it NO DON'T YOU DARE 🙃.

I tried surefire and surefire + failsafe, doesnt change crap. 

Plz help, or I'm going to rewrite the lecture in go. 😄

**Comment by jlevesy** [comment]
@dduportal made the changes we discussed this morning. This PR is now ready to review 🎉

File: content/chapitres/tests.adoc [hunk 1]
```diff
@@ -6,10 +6,10 @@ C'est du code qui vérifie que du code fait ce qu'il est supposé faire.
 
 == Pourquoi faire des tests ?
 
-* Ça sert a prouver que le logiciel se comporte comme attendu a tout moment.
+* Prouve que le logiciel se comporte comme attendu a tout moment.
 * Détecte les impacts non anticipés des changements introduits
 * Evite l'introduction de régressions
-* Écrire des tests est un acte préventif et non curatif.
+* Ecrire des tests est un acte préventif et non curratif.
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Écrire des tests est un acte préventif et non curatif.
```

File: content/chapitres/tests.adoc [hunk 2]
```diff
@@ -279,20 +279,90 @@ Tester des composants indépendamment ne prouve pas que le système fonctionne u
...
+== Définition du SUT (1/2)
 
 image::architecture.png[]
 
-== Configuration de Maven
+== Définition du SUT (2/2)
+
+Un test d'intégration doit a chaque test:
+
+* Démarer et provisionner un environement d'execution (une DB, Elasticsearch, un autre service...)
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* DémaRrer et provisionner un environnement d’exécution (une DB, Elasticsearch, un autre service...)
```

File: content/chapitres/tests.adoc [hunk 3]
```diff
@@ -279,20 +279,90 @@ Tester des composants indépendamment ne prouve pas que le système fonctionne u
...
+* Démarer votre application
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Démarrer votre application
```

File: content/chapitres/tests.adoc [hunk 4]
```diff
@@ -279,20 +279,90 @@ Tester des composants indépendamment ne prouve pas que le système fonctionne u
...
+* Jouer un scenario de test
+* Eteindre et nettoyer son environement d'execution pour garantir l'isolation des tests
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Éteindre et nettoyer son environnement d’exécution pour garantir l'isolation des tests
```

File: content/chapitres/tests.adoc [hunk 5]
```diff
@@ -279,20 +279,90 @@ Tester des composants indépendamment ne prouve pas que le système fonctionne u
...
 image::architecture.png[]
 
-== Configuration de Maven
+== Définition du SUT (2/2)
+
+Un test d'intégration doit a chaque test:
+
+* Démarer et provisionner un environement d'execution (une DB, Elasticsearch, un autre service...)
+* Démarer votre application
+* Jouer un scenario de test
```
Comment (dduportal) [inline content/chapitres/tests.adoc]:
```suggestion
* Jouer un scénario de test
```

**Review by dduportal — APPROVED** [review]
Language but it's cool otherwise!

I've started playing around with the exercises to reproduce them on the current git pod, but haven't finish yet. WDYT if we stil merge and then fix errors in exercises on the go (if we find some)?

**Comment by jlevesy** [comment]
Agreed, we should meet and review the whole lecture together :)

**Pull request merged**